### PR TITLE
Refactor print_results() function

### LINKED ISSUE #275: Lens print_results() fails
## Expected Behavior
`print_results()` in `Lens` class should successfully print results.

## Actual Behavior
Error:
image

Seems to have been introduced when structure of `get_results()` return object changed in #214

Minimal example:

```
from credoai.lens import Lens
from credoai.artifacts import ClassificationModel, TabularData
from credoai.datasets import fetch_credit_model
from credoai.evaluators import Performance

(
    X_train,
    y_train,
    sensitive_features_train,
    X_test,
    y_test,
    sensitive_features_test,
    model,
) = fetch_credit_model(True)

# Set up model and data artifacts
credo_model = ClassificationModel(
    name="credit_default_classifier",
    model_like=model,
    tags={"model_type": "binary_classification"},
)

credo_test = TabularData(
    name="UCI-credit-test",
    X=X_test,
    y=y_test,
    sensitive_features=sensitive_features_test,
)

credo_train = TabularData(
    name="UCI-credit-train",
    X=X_train,
    y=y_train,
    sensitive_features=sensitive_features_train,
)

# Initialization of the Lens object
lens = Lens(
    model=credo_model,
    assessment_data=credo_test,
    training_data=credo_train,
)

lens.add(Performance(["accuracy_score"]))

# Run all the Assessments
lens.run()

lens.print_results()

```

### PR DESCRIPTION
**Pull request by esherman-credo**
## Describe your changes
Modify `print_results()` function to reflect new format of return object from `get_results()`. Metadata and results are separate per evaluator per assessment.

## Issue ticket number and link
Fixes #275 

## Known outstanding issues that are not fully accounted for
None

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have built basic tests for new functionality (particularly new evaluators)
- See #272 
- [x] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctl
- N/A
- [x] Will this be part of a major product update? If yes, please write one phrase about this update.
- No

## Extra-mile Checklist
- [x] I have thought expansively about edge cases and written tests for them
- N/A

### CODE REVIEW

File: credoai/lens/lens.py [hunk 1]
```diff
@@ -303,10 +303,11 @@ def get_results(self, evaluator_name=None, metadata=None) -> List[Dict]:
...
-        for key, val in results.items():
-            print(f"Evaluator: {key}\n")
-            for i in val:
-                print(i)
+        for result_grouping in results:
+            for key, val in result_grouping["metadata"].items():
+                print(f"{key.capitalize()}: {val}")
+            for val in result_grouping["results"]:
+                print(val)
                 print()
```
Comment (fabrizio-credo) [inline credoai/lens/lens.py:311]: You can use `print(f"{val}\n")`, instead of the empty `print()`.

**Review by fabrizio-credo — APPROVED** [review]
